WcaOnRails/app/models/team.rb

```diff
@@ -25,11 +25,11 @@ def membership_periods_cannot_overlap_for_single_user
 
   # Code duplication from Cachable concern, as we index by friendly_id and not by id :(
   def self.c_all_by_friendly_id
-    @@teams_by_friendly_id ||= all.index_by(&:friendly_id)
+    @@teams_by_friendly_id ||= all.unscoped.index_by(&:friendly_id)
```

Comment:
*Idea:* create an additional scope like:
```ruby
scope :with_hidden, -> { unscope(:where) }
```
Then we would query visible teams implicitly, and **all** teams with `Team.with_hidden`.

Reply:
Good idea, done!